File: src/main/java/io/anontech/vizivault/VaultException.java
package io.anontech.vizivault;

public class VaultException extends RuntimeException {

  private static final long serialVersionUID = 1L;

  private final String message;
  private final int statusCode;

  public VaultException(String message, int status) {
    this.message = message;
    this.statusCode = status;
  }

  @Override
  public String getMessage() {
    return message;
  }

  public int getStatus() {
    return statusCode;
  }

}

File: src/VaultException.ts
export class VaultException extends Error {
  private statusCode: number;
  public message: string;

  public constructor(message: string, status: number) {
    super();
    this.message = message;
    this.statusCode = status;
  }

  public getStatusCode(): number {
    return this.statusCode;
  }

  public setStatusCode(statusCode: number): void {
    this.statusCode = statusCode;
  }
}
